### PR TITLE
UICM-98: Timezones, Update the datepicker and UiFramework version

### PR DESCRIPTION
**Pull request by icrc-jfigueiredo**
#### Ticket:
https://issues.openmrs.org/browse/UICM-98
#### What I have done:
* Update **datetimepicker.gsp** to support time zones.
* Update UIFR version from 3.8 to 3.21.0-SNAPSHOT to access the new `UiUtils` time zones API.

### CODE REVIEW

**Comment by mogoodrich** [comment]
Was going to review, but saw it was still in draft... feel free to ping me when it's ready...

File: omod/src/main/webapp/fragments/field/datetimepicker.gsp [hunk 1]
```diff
@@ -39,8 +39,13 @@
     def defaultDateString = ""
     def defaultDateISOFormatted = ""
     if (defaultDate) {
-        defaultDateString = dateStringFormat.format(defaultDate)
-        defaultDateISOFormatted = dateISOFormatted.format(defaultDate)
+        if( ui.handleTimeZones() ) {
```
Comment (mks-d) [inline omod/src/main/webapp/fragments/field/datetimepicker.gsp:42]: → `ui.convertTimezones()`

File: omod/src/main/webapp/fragments/field/datetimepicker.gsp [hunk 2]
```diff
@@ -120,11 +125,19 @@
         format: "${datePickerFormat}",
 
         <% if (startDate) { %>
-            startDate: "${ startDate instanceof Date ? dateISOFormatted.format(startDate) : startDate }",
+            <% if (ui.handleTimeZones()) { %>
```
Comment (mks-d) [inline omod/src/main/webapp/fragments/field/datetimepicker.gsp:128]: → `ui.convertTimezones()`

File: omod/src/main/webapp/fragments/field/datetimepicker.gsp [hunk 3]
```diff
@@ -39,8 +39,13 @@
     def defaultDateString = ""
     def defaultDateISOFormatted = ""
     if (defaultDate) {
-        defaultDateString = dateStringFormat.format(defaultDate)
-        defaultDateISOFormatted = dateISOFormatted.format(defaultDate)
+        if( ui.handleTimeZones() ) {
+            defaultDateString = ui.format(defaultDate)
+            defaultDateISOFormatted = ui.format(defaultDate)
+        }else{
```
Comment (mks-d) [inline omod/src/main/webapp/fragments/field/datetimepicker.gsp:45]: → `} else {` (spacing)

File: omod/src/main/webapp/fragments/field/datetimepicker.gsp [hunk 4]
```diff
@@ -120,11 +125,19 @@
         format: "${datePickerFormat}",
 
         <% if (startDate) { %>
-            startDate: "${ startDate instanceof Date ? dateISOFormatted.format(startDate) : startDate }",
+            <% if (ui.handleTimeZones()) { %>
+                startDate: "${ startDate instanceof Date ? ui.format(startDate) : startDate }",
+            <% }else{ %>
```
Comment (mks-d) [inline omod/src/main/webapp/fragments/field/datetimepicker.gsp:130]: → `} else {` (spacing)

File: omod/src/main/webapp/fragments/field/datetimepicker.gsp [hunk 5]
```diff
@@ -120,11 +125,19 @@
...
+                startDate: "${ startDate instanceof Date ? dateISOFormatted.format(startDate) : startDate }",
+            <% } %>
         <% } %>
 
         <% if (endDate) { %>
-            endDate: "${ endDate instanceof Date ? dateISOFormatted.format(endDate) : endDate }",
+            <% if (ui.handleTimeZones()) { %>
+                endDate: "${ endDate instanceof Date ? ui.format(endDate) : endDate }",
+            <% }else{ %>
```
Comment (mks-d) [inline omod/src/main/webapp/fragments/field/datetimepicker.gsp:138]: → `} else {` (spacing)

**Review by mks-d — CHANGES_REQUESTED** [review]

**Pull request merged**